src/main/scala/scorex/core/network/Handshake.scala

```diff
@@ -34,15 +31,26 @@ object HandshakeSerializer extends Serializer[Handshake] {
 
     val nodeNameBytes = obj.nodeName.getBytes
 
-    Array(anb.size.toByte) ++ anb ++
+    val featureBytes = obj.features.foldLeft(Array(obj.features.size.toByte)){case (fb, f) =>
+      val featId = f.featureId
+      val featBytes = f.bytes
+      fb ++ (featId +: Shorts.toByteArray(featBytes.length.toShort)) ++ featBytes
+    }
+
+    Array(anb.size.toByte) ++
```

Comment:
For performance reasons it's better to use `scorex.core.utils.concatBytes` or `com.google.common.primitives.Bytes.concat` instead of `++`